File: backend/app/schemas/page.py
from datetime import datetime
from typing import Optional

from pydantic import BaseModel, ConfigDict


class PageConnect(BaseModel):
    fb_page_id: str
    name: str
    access_token: str
    category: Optional[str] = None


class PageResponse(BaseModel):
    model_config = ConfigDict(from_attributes=True)

    id: str
    fb_page_id: str
    name: str
    category: Optional[str] = None
    token_status: str = "configured"
    queued_count: int = 0
    last_published_at: Optional[datetime] = None
    created_at: Optional[datetime] = None


class PageListResponse(BaseModel):
    pages: list[PageResponse]

File: frontend/src/types/page.ts
export type Platform = 'facebook' | 'telegram' | 'twitter'

export interface Page {
  id: string
  fb_page_id: string
  name: string
  category?: string
  platform?: Platform
  token_status: 'configured' | 'missing'
  queued_count: number
  last_published_at?: string
  created_at?: string
}

export interface PageConnect {
  fb_page_id: string
  name: string
  access_token: string
  category?: string
  platform: Platform
}
